core/common/src/main/java/alluxio/underfs/NonAtomicFileOutputStream.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.underfs;
+
+import alluxio.underfs.options.NonAtomicCreateOptions;
+
+import java.io.FilterOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+
+import javax.annotation.concurrent.NotThreadSafe;
+
+/**
+ * An {@link NonAtomicFileOutputStream} writes to a temporary file and renames on close.
+ */
+@NotThreadSafe
+public class NonAtomicFileOutputStream extends FilterOutputStream {
```

Comment:
Changed to using OutputStream and writing implementation for write calls. FilterOutputStrem provides default implementations for all OutputStream calls, but that was causing the test case failures.